src/Cache.zig

```diff
@@ -508,8 +508,10 @@ pub const Manifest = struct {
 
         // Next we will check the globally shared Cache timestamp, which is accessed
         // from multiple threads.
-        man.cache.mutex.lock();
-        defer man.cache.mutex.unlock();
+        if (builtin.os.tag != .wasi or !builtin.single_threaded) { // WASI does not support flock
```

Comment:
Should this not be

```suggestion
        if (builtin.os.tag != .wasi and !builtin.single_threaded) { // WASI does not support flock
```

If either WASI or single-threaded is true, we shouldn't need to lock anything, am I right?

Reply:
I think this is a mistake on my part - I probably thought this was locking the manifest file, but we can actually leave the code as-is. The standard library already ensures Mutex lock/unlock is a noop in single-threaded mode 🙂 

Pushing an update now.

(While at it, I also cleaned up the file locks to use `can_spawn` to determine whether `flock` is needed to contend with multiple processes.)

Reply:
Oh nice one! Thanks for investigating further and updating the PR!